libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -591,6 +581,56 @@ static uint8_t * _encodeUserName( uint8_t * pDestination,
 
 /*-----------------------------------------------------------*/
 
+static uint8_t * _encodeUserNameAndMetrics( uint8_t * pDestination,
```

Comment:
I know this PR is for reorganization, however I want to make sure we take all opportunities to improve robustness of this code.  This function isn't taking a parameter for how much space is available after `pDestination`, yet that [information is available if we go up the call stack a bit](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/iot_mqtt_serialize.c#L1414).  Note that there is already a bounds check on line 595, if you had just carried the available packet size down to this function you could use that and be totally safe, it wouldn't be hard to add.  Lines 610/617 are frightening without it.

Reply:
This function and `_encodeUserName` are similar to `_encodeString` and `_encodeRemainingLength` in that the size of the buffer is always checked before they are called, and so don't check it themselves. However, both of those functions have some sort of warning:
```
 @warning This function does not check the size of `pDestination`! Ensure that
 `pDestination` is large enough to hold `sourceLength+2` bytes to avoid a buffer
 overflow.
```
Would adding a similar warning to these functions suffice?

Reply:
At a minimum let's go ahead and add that warning to be consistent.